airbyte-workers/src/main/java/io/airbyte/workers/protocols/airbyte/NamespacingMapper.java

```diff
@@ -41,39 +43,59 @@
  */
 public class NamespacingMapper implements Mapper<AirbyteMessage> {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(NamespacingMapper.class);
+
   private final NamespaceDefinitionType namespaceDefinition;
-  private final String prefix;
+  private final String namespaceFormat;
+  private final String streamPrefix;
 
-  public NamespacingMapper(NamespaceDefinitionType namespaceDefinition, String prefix) {
+  public NamespacingMapper(NamespaceDefinitionType namespaceDefinition, String namespaceFormat, String streamPrefix) {
     this.namespaceDefinition = namespaceDefinition;
-    this.prefix = prefix;
+    this.namespaceFormat = namespaceFormat;
+    this.streamPrefix = streamPrefix;
   }
 
   @Override
   public ConfiguredAirbyteCatalog mapCatalog(final ConfiguredAirbyteCatalog inputCatalog) {
     final ConfiguredAirbyteCatalog catalog = Jsons.clone(inputCatalog);
-    catalog.getStreams().forEach(s -> mutateStream(s.getStream(), namespaceDefinition, prefix));
+    catalog.getStreams().forEach(s -> {
+      final AirbyteStream stream = s.getStream();
+      if (namespaceDefinition != null) {
```

Comment:
suggestion to simplify the if condition.

```
if (namespaceDefinition == null) {
   stream.withName(transformStreamName(stream.getName(), streamPrefix));
}

if (namespaceDefinition.equals(NamespaceDefinitionType.DESTINATION)) {
  stream.withNamespace(null);
} 

if (namespaceDefinition.equals(NamespaceDefinitionType.CUSTOMFORMAT)) {
    stream.withNamespace(formatNamespace(stream.getNamespace(), namespaceFormat));
}
```

Reply:
the mutation to stream name is always happening regardless of namespace configurations.

In your suggestion, doesn't it end in Nullpointer exceptions?